### PR TITLE
added test for application launcher

### PR DESCRIPTION
**Pull request by pnaik1**
Closes: https://issues.redhat.com/browse/RHOAIENG-4511

## Description
added test for application Launcher

## How Has This Been Tested?
npm run test

## Request review criteria:

Self checklist (all need to be checked):
- [x] The developer has manually tested the changes and verified that the changes work
- [x] Commits have been squashed into descriptive, self-contained units of work (e.g. 'WIP' and 'Implements feedback' style messages have been removed)
- [x] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [ ] The developer has added tests or explained why testing cannot be added (unit or cypress tests for related changes)

If you have UI changes: 

- [ ] Included any necessary screenshots or gifs if it was a UI change.
- [ ] Included tags to the UX team if it was a UI/UX change (find relevant UX in the [SMEs](https://github.com/opendatahub-io/odh-dashboard/tree/main/docs/smes.md) section).

After the PR is posted & before it merges:
- [ ] The developer has tested their solution on a cluster by using the image produced by the PR to `main`

### CODE REVIEW

File: frontend/src/__tests__/cypress/cypress/e2e/appLauncher/AppLauncher.cy.ts [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+describe('App launcher', () => {
+  beforeEach(() => {
+    cy.intercept('/api/dsc/status', mockDscStatus({}));
+    cy.intercept('/api/status', mockStatus());
+    cy.intercept('/api/config', mockDashboardConfig({}));
+    cy.intercept('/api/components', mockComponents());
+    applicationLauncher.visit();
+  });
+  it('Validate clicking on App Launcher opens menu', () => {
+    applicationLauncher.visit();
```
Comment (christianvogt) [inline frontend/src/__tests__/cypress/cypress/e2e/appLauncher/AppLauncher.cy.ts:14]: Remove duplicate `visit`.

File: frontend/src/__tests__/cypress/cypress/pages/appLauncher.ts [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+}
+
+class ApplicationLauncher {
+  visit() {
+    cy.visit('/explore');
+    this.wait();
+  }
+
+  private wait() {
+    cy.findByTestId('explore-applications').should('be.visible');
```
Comment (christianvogt) [inline frontend/src/__tests__/cypress/cypress/pages/appLauncher.ts:17]: Wait for something related to the chrome and not the page contents.

File: frontend/src/__tests__/cypress/cypress/pages/appLauncher.ts [hunk 3]
```diff
@@ -0,0 +1,34 @@
+import { Contextual } from './components/Contextual';
+
+class ApplicationLauncherMenuGroup extends Contextual<HTMLElement> {
+  shouldHaveApplicationLauncherItem(name: string) {
+    this.find().findAllByTestId('application-launcher-item').contains(name).should('exist');
+    return this;
+  }
+}
+
+class ApplicationLauncher {
```
Comment (christianvogt) [inline frontend/src/__tests__/cypress/cypress/pages/appLauncher.ts:10]: I forgot that we have a page object for the chrome.
Move to `appChrome.ts`

File: frontend/src/__tests__/cypress/cypress/pages/appLauncher.ts [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+  visit() {
+    cy.visit('/explore');
+    this.wait();
+  }
```
Comment (christianvogt) [inline frontend/src/__tests__/cypress/cypress/pages/appLauncher.ts:14]: Move this to function to `AppChrome` object and make it visit the root of the app.

```suggestion
  visit() {
    cy.visit('/');
    this.wait();
  }
```

File: frontend/src/__tests__/cypress/cypress/pages/appLauncher.ts (hunk 3 above)
Comment (christianvogt) [inline frontend/src/__tests__/cypress/cypress/pages/appLauncher.ts:10]: Add a `getApplicationLauncher` function to `AppChrome`

**Comment by pnaik1** [comment]
Updated the pr changes

**Pull request merged**